trieste/models/model_interfaces.py

```diff
@@ -557,6 +607,48 @@ def update(self, dataset: Dataset, *, jitter: float = DEFAULTS.JITTER) -> None:
         model.q_mu = gpflow.Parameter(new_q_mu)
         model.q_sqrt = gpflow.Parameter(new_q_sqrt, transform=gpflow.utilities.triangular())
 
+    def optimize(self, dataset: Dataset) -> None:
+        """
+        :class:`VariationalGaussianProcess` has a custom `optimize` method that (optionally) permits
+        alternating between standard optimization steps (for kernel parameters) and natural gradient
+        steps for the variational parameters (`q_mu` and `q_sqrt`). See :cite:`salimbeni2018natural`
+        for details. Using natural gradients can dramatically speed up model fitting, especially for
+        ill-conditioned posteriors.
+
+        If using natural gradients, our optimizer inherits the mini-batch behavior and number
+        of optimization steps as the base optimizer specified when initializing
+        the :class:`VariationalGaussianProcess`.
+        """
+        model = self.model
+
+        if self._use_natgrads:  # optimize variational params with natgrad optimizer
+
+            natgrad_optimizer = gpflow.optimizers.NaturalGradient(gamma=self._natgrad_gamma)
+            base_optimizer: TFOptimizer = self.optimizer
```

Comment:
mypy can't 'remember' that this was checked previously. Either use `cast` here or assign it to a different variable `self.tfoptimizer` during initialization.

Reply:
@uri-granta what do you mean by cast?